### PR TITLE
Final Capstone Project

### PR DESCRIPTION
**Pull request by IndieCoderMM**
# Building a Full-stack E-Learning Platform 

In this capstone project, we built a website that allows users to reserve online classes while also providing the opportunity to schedule study sessions with fellow learners in their area. 

## 💻 Implemented Features
- Users can log in or sign up by typing their username
- We followed the layout of [this template design](https://www.behance.net/gallery/26425031/Vespa-Responsive-Redesign)
- Implemented Pages:
  - Home: displays all available courses
  - Detail: displays detailed information about a course
  - Reservation: displays the list of reserved courses
  - New Course: form to create a new course
  - New Reservation: form to create a new reservation
  - Delete Course: displays courses with delete buttons
- API endpoints:
  - *Course*: `[:index :show :create :destroy]`
  - *Reservation*: `[:index :create :destroy]`
- Responsive design: Layout for both desktop and mobile
- Tests: *model_specs*, *request_specs* and *unit_tests*
- API documentation available on `/api-docs/`

## 🔗 Links 
- [Live Demo Link](https://classup.onrender.com/)
- [Link to Kanban Board](https://github.com/users/IndieCoderMM/projects/4)

*Thanks for reviewing our project. We'd love to hear your feedback*

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,24 +1,247 @@
...
+
+To read the API documentation, visit `https://localhost:3000/api-docs`
+
+### 🧪 Tests
+
+To run tests, run the following command:
+
+```sh
+rspec spec --format doc
+```
```
Comment (Meltrust) [inline README.md:176]: - [ ]  You are using Rspec, but the "test" folder still exists, which is used by another test framework. 

image


Kindly consider deleting that folder if you are not using it. 👍

File: app/javascript/components/App.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import NewCourse from '../pages/NewCourse';
+import DeleteCourse from '../pages/DeleteCourse';
+import NewReservation from '../pages/NewReservation';
+import SideNavbar from './SideNavbar';
+import Login from '../pages/Login';
+import Detail from '../pages/Details';
+import Header from './Header';
+import Profile from '../pages/Profile';
+
+function App() {
```
Comment (Meltrust) [inline app/javascript/components/App.js:16]: - [ ]  Please, kindly refactor this and all other functions in your project (5) to ES6 arrow functions, so your code complies with the latest standards.

Refactoring is easy:

```javascript
const App = () => {
  // ...
}
```

## Explanation

The `function` keyword has been soft deprecated in almost all the software industry in favor of ES6 arrow functions. This means, almost nobody uses it anymore, so much that it's considered a non-best practice.

You can do a search in Vscode for the "function" word, and correct as needed (I found 5).  Shouldn't take you more than 5 min so don't worry:

image

That way, your code will meet the latest standards 💪

**Review by Meltrust — CHANGES_REQUESTED** [review]
Hi @IndieCoderMM,

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation, but you are almost there!






### To highlight:

- Nice code organization✔️ 
- Backend tests are passing✔️ 
- Frontend tests are passing✔️
- Nice frontend implementation✔️ 
- Really nice mobile implementation✔️ 
- Nice model validations ✔️ 
- Nice documentation ✔️ 
- Really good readme ✔️ 
- App is working well ✔️ 



#### You are really close to finishing the Microverse program!! Keep it up! 👍👍👍

You can do it

### After implementing the requested changes, please submit another review request. ♻️
_Check the comments under the review._







Cheers and Happy coding!👏👏👏

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the previous reviews unless it is requested otherwise._**

------

File: app/javascript/components/App.js (hunk 2 above)
Comment (IndieCoderMM) [reply app/javascript/components/App.js:16]: Thanks so much for your very detailed and insightful review. 🙏 
We truly appreciate it and we'll be implementing all the required changes.

**Review by Meltrust — APPROVED** [review]
#### Hi @IndieCoderMM,


# **Wow, you did it 🎉**
Brilliant
## Thank you for the changes implemented 💪 🥇 ㊗️ 
#### _Unless you want to add more features, go ahead to your final presentation ⏩ ⏩ ⏩_ 
## You are about to finish the Microverse program.  You have come a long way!!!




### Good luck in the software industry!!  I'll see you there. ✨
### **Congratulations!!!!!!**  🎉
applause

### To highlight

- Superfluous 'test' folder has been removed✔️ 
- All JS functions are now ES6✔️ 
- Awesome job✔️ 



Cheers and Happy coding!👏👏👏


------

**Pull request merged**